packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -1225,6 +1234,312 @@ export class ProfileInfo {
+     * @returns {boolean} `true` if added to the loaded config; `false` otherwise
+     */
+    public addProfileToConfig(profileType: string, layerPath?: string): boolean {
+        // Find the schema in the cache, starting with the highest-priority layer and working up
+        const profileSchema = [...this.getTeamConfig().mLayers].reverse()
+            .reduce((prev: IProfileSchema, cfgLayer) => {
+                const cachedSchema = [...this.mProfileSchemaCache.entries()]
+                    .filter(([typeWithPath, schema]) => typeWithPath.includes(`${cfgLayer.path}:${profileType}`))[0];
+                if (cachedSchema != null) {
+                    prev = cachedSchema[1];
+                }
+                return prev;
+            }, undefined);
+
+        // Skip adding to config if the schema was not found
+        if (profileSchema == null) {
+            return false;
+        }
+
+        this.getTeamConfig().api.profiles.set(layerPath ? `${layerPath}.${profileType}` : profileType,
+            ConfigBuilder.buildDefaultProfile({ type: profileType, schema: profileSchema }, { populateProperties: true }));
+        return true;
+    }
+
+    /**
+     * Updates the schema to contain the new profile type.
+     * If the type exists in the cache, it will use the matching layer; if not found, it will use the schema at the active layer.
+     *
+     * @param {string} profileType The profile type to add into the schema
+     * @param {IProfileSchema} typeSchema The schema for the profile type
+     * @param [versionChanged] Whether the version has changed for the schema (optional)
+     * @returns {boolean} `true` if added to the schema; `false` otherwise
+     */
+    private updateSchemaAtLayer(profileType: string, schema: IProfileSchema, versionChanged?: boolean): void {
+        // Check if type already exists in schema cache; if so, update schema at the same layer.
+        // Otherwise, update schema at the active layer.
+        const cachedType = [...this.mProfileSchemaCache.entries()]
+            .find(([typePath, _schema]) => typePath.includes(`:${profileType}`));
+
+        const layerPath = cachedType != null ? cachedType[0].substring(0, cachedType[0].lastIndexOf(":")) : this.getTeamConfig().layerActive().path;
+        const layerToUpdate = this.getTeamConfig().mLayers.find((l) => l.path === layerPath);
+        const cacheKey = `${layerPath}:${profileType}`;
+
+        const sameSchemaExists = versionChanged ? false :
+            this.mProfileSchemaCache.has(cacheKey) && lodash.isEqual(this.mProfileSchemaCache.get(cacheKey), schema);
+        // Update the cache with the newest schema for this profile type
+        this.mProfileSchemaCache.set(cacheKey, schema);
+
+        const schemaUri = new url.URL(layerToUpdate.properties.$schema, url.pathToFileURL(layerPath));
+        const schemaPath = url.fileURLToPath(schemaUri);
```

Comment:
I understand we haven't talked much about supporting the use-case of hosting the schema file somewhere.
However, I'm curious if we should do something similar to the `Config#getSchemaInfo()` function.
- https://github.com/zowe/zowe-cli/blob/master/packages/imperative/src/config/src/Config.ts#L380

Reply:
Ah, I didn't realize this function existed! 😅 

After today's discussion about only having to update the global-level schema, it probably would make more sense to use `getSchemaInfo`. I will test this out locally to see if this changes the behavior first, but I think that would be a nice change.

Reply:
I'm not sure that we can do much about updating a schema located at a web URL. I feel like that would require some extensive work to support connecting over some interface (SSH, FTP, etc.) or an API to update the contents of a remote schema.

That being said, I added a check that will not update the schema when the built URL does not have the `file` protocol. If it is the `file` protocol, even mangled paths such as `file:///Users/traeok/https:/a.txt` will work fine with `existsSync` (and will just return false).